bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureBridgeHandler.java

```diff
@@ -0,0 +1,304 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.verisure.internal.handler;
+
+import static org.openhab.binding.verisure.internal.VerisureBindingConstants.*;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.verisure.internal.DeviceStatusListener;
+import org.openhab.binding.verisure.internal.VerisureBridgeConfiguration;
+import org.openhab.binding.verisure.internal.VerisureSession;
+import org.openhab.binding.verisure.internal.discovery.VerisureThingDiscoveryService;
+import org.openhab.binding.verisure.internal.model.VerisureThing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VerisureBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author l3rum - Initial contribution
+ * @author Jan Gustafsson - Furher development
+ */
+@NonNullByDefault
+public class VerisureBridgeHandler extends BaseBridgeHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private static final int REFRESH_DELAY_SECONDS = 30;
+    private final Logger logger = LoggerFactory.getLogger(VerisureBridgeHandler.class);
+    private final ReentrantLock immediateRefreshJobLock = new ReentrantLock();
+
+    private String authstring = "";
+    private @Nullable String pinCode;
+    private static int refresh = 600;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> immediateRefreshJob;
+    private @Nullable VerisureSession session;
+    private final HttpClient httpClient;
```

Comment:
Please move this field above the non-final fields.

Reply:
Fixed!